### ISSUE TITLE
Instrument Plugins Redesign

### ISSUE THREAD

**Issue by RebeccaDeField** [LMMS/lmms#3534]
We are going to be enhancing the instrument designs. I'm looking for a balance between an exciting ui (letting each plugin have its own unique style to represent and visually distinguish its sound) and good ux (maintaining some consistency so that the user does not become surprised/confused and understands that these plugins have relation to LMMS.

Bellow, I have listed all of the elements and design standards that will be need to be followed by the people contributing designs. If someone would like to contribute a design for one of the plugins, please inform me in the comments bellow so that two people do not accidentally redesign the same plugin.

**FONTS**
When working on the logos, you can pick the font of your choice as long as the font has proper licensing. Some safe places to find fonts are google.com/fonts and fontsquirrel.com. The fonts that we are currently using for LMMS include Orbitron in ALL CAPS, Titilliumm, Cabin and Open Sans. For best results when working with text, export your final SVG as a Cairo .png (I prefer to use the Save A Copy function here).

If the Cairo .png leaves some of the elements looking a bit too crisp, you can always export the text with Cairo and the background with Inkscape. Use the "Color to Alpha" option in GIMP to cut out the bg and then place it on the background. This is only necessary if you are creating a more complex graphic.

**BACKGROUND**
We want the background to have enough contrast, especially with the knobs.

**GRADIENTS**
Please note that your eyes follow the direction that a gradient is facing. For example, the gradient that I used in the LFO controller was linear so that your eyes would be drawn from the title down to the content of the plugin. If the gradient pointing at an odd angle, the plugin will be a bit less intuitive for the user. Gradients should be subtle when possible. This is not exactly a rule, but because many prefer flat design and many others don't I try to keep it somewhere in between.

**KNOBS**
Let's keep the design of the knobs the same throughout the instruments with the exception of choosing any of the colors in our palette.

**COLORS**
Unlike the effects plugins, which almost exclusively used the green, you can use any of the colors in this palette on the instrument designs.
![colors](https://cloud.githubusercontent.com/assets/7960169/25774816/c40999be-324b-11e7-9d3b-b7052fde8c47.png)

Please reference my source files in the [artwork repo](https://github.com/LMMS/artwork) for access to backgrounds, buttons, and icons that you can reuse.

Note: you can replace @mentions with PR # once ready.

- [ ] AudioFileProcessor
- [ ] BitInvader
- [ ] FreeBoy
- [ ] GIG Player
- [ ] Kicker
- [ ] LB302
- [ ] Mallets
- [ ] Monstro
- [ ] Nescaline
- [ ] OpulenZ
- [ ] Organic
- [ ] PatMan
- [ ] Sf2 Player
- [ ] sfxr
- [ ] SID
- [ ] TripleOscilator
- [ ] VesSTige
- [ ] Vibed
- [ ] Watsyn
- [ ] ZynAddSubFX
- [ ] Carla Rack
- [ ] Carla Patchbay
- [ ] Xpressive (when completed)

![shot-2017-05-06_11-22-38](https://cloud.githubusercontent.com/assets/7960169/25774938/216f8d8c-324e-11e7-8e3e-d878cf894cbc.png)

@Umcaruje @Mark-Agent003 @BaraMGB

**Comment by Texxiliator** [comment]
So basically you are asking if people could make a new desing for the synths?
Ok could i do Monstro? (If i got something wrong sorry)

**Comment by RebeccaDeField** [comment]
@Texxiliator thanks for talking to me on Discord. I have added you to the Monstro plugin. Feel free to let me know if you have any questions.

**Comment by JordanKyser** [comment]
I could take the ZynAddSubFX plugin if that's okay with you. I will try my best to re-design (I am pretty much saying what Texxiliator said).

**Comment by IvanMaldonado** [comment]
I onced started a issue (#2404) proposing an artwork for the organic plugin and I was implyied it needed some specific reasons to change an artwork, and that redisigning was more than just re-skinning, I also had a lot of difficulties trying to convince some people here that changes like these could be a good a idea.

I'm happy to see things are changing here.

**Comment by JordanKyser** [comment]
@IvanMaldonado I agree that the Organic plug-in needs redesigning. This also goes for the AudioFileProcessor, Patman (who uses this?), GIG (and this?), SID, ZynAddSubFX (in which I just asked to volunteer to re-design this)

**Comment by JordanKyser** [comment]
@RebeccaLaVie In case you didn't hear yet, I have request ZynAddSubFX redesign and IvanMaldonado has requested to redesign Organic

**Comment by Umcaruje** [comment]
>This also goes for the AudioFileProcessor, Patman (who uses this?), GIG (and this?), SID, and ZynAddSubFX (in which I just asked to volunteer to re-design this) plugins
>
>@RebeccaLaVie In case you didn't hear yet, I have requested to redesign ZynAddSubFX and IvanMaldonado has requested to redesign Organic

@JordanKyser relax. This project will span over months, and all of us read comments, but not instantly, so please just be patient. No need for repeating youself. Also if there's no name next to a plugin, nobody is currently assigned to it, but I would not suggest taking on more than one plugin per person at a time.

I will assign you and @IvanMaldonado to the respective instruments.

**Comment by JordanKyser** [comment]
> @JordanKyser relax. This project will span over months, and all of us read comments, but not instantly, so please just be patient. No need for repeating youself. Also if there's no name next to a plugin, nobody is currently assigned to it, but I would not suggest taking on more than one plugin per person at a time.

I will assign you and @IvanMaldonado to the respective instruments

I am stating plugins that have not have redesigning for the past 3 or more updates.

**Comment by Umcaruje** [comment]
> I am stating plugins that have not have redesigning for the past 3 or more updates. Also, it has been a long while since 1.1.3 was released, so my guess is that there are many others that are waiting for the stable 1.2 release.

The redesign of the instrument plugins does not affect the 1.2 release. Also all unchecked instruments are a subject to redesign.

**Comment by RebeccaLaVie** [comment]
> I once started an issue (#2404) proposing an artwork for the organic plugin

I was going to reach out to you about working with my on the organic plugin, so I'm glad to see that you're interested. 

> I also had a lot of difficulties trying to convince some people here that changes like these could be a good idea.

Not only have I read over your comments regarding this, but I have also supportive of all your efforts. We weren't ready for a full redesign project while the effect plugins were incomplete and I am still working on the last effects plugin.

These sort of redesign projects take time. All of your comments will be read and all of your feedback will be considered :+1:

**Comment by Hussam-Eddin-Alhomsi** [comment]
@RebeccaLaVue
I'm new to LMMS and I have a question.
Can LMMS's instrument plug-ins be used in other DAWs?
If they can, I agree that with you on "letting each plug-in have its own unique design to represent and visually distinguish its sound".
But if they are only to be used inside LMMS, I think that they should be redesigned around a main theme that is similar to LMMS without too much variation.

**Comment by Texxiliator** [comment]
@Hussam-Eddin-Alhomsi i think you cant use any instrument in other DAW (other then vst,soundfont and so) they are only in lmms. And thats what im trying to go for. Look like it is part of Lmms.

**Comment by Mark-Agent003** [comment]
That's what we are trying to do now, we are trying to make the instruments follow the same theme, look like they are part of LMMS, and also:
>letting each plug-in have its own unique design to represent and visually distinguish its sound

Currently the instrument visuals are outdated, so that's what this is. 😄

**Comment by Hussam-Eddin-Alhomsi** [comment]
@Mark-Agent003 Thank you for clarifying. :)

**Comment by Mark-Agent003** [comment]
I think it goes without saying, but I will work on the SF2 Player.

**Comment by Texxiliator** [comment]
@Mark-Agent003 Wow you are really good ! :D

**Comment by tresf** [comment]
> Can LMMS's instrument plug-ins be used in other DAWs?

No and there are no plans to be unless they are rewritten as LV2 plugins.

**Comment by RebeccaLaVie** [comment]
> letting each plug-in have its own unique design to represent and visually distinguish its sound

@Mark-Agent003 By this, I mean that anyone can use any of the colors in the palette above and a custom logo to add a little bit of slightly distinguish each plugin, because this has been requested by the community. We still want the knobs, colors, and style to stay the same.

> I think it goes without saying, but I will work on the SF2 Player.

I already added you to the sf2 player. :+1:

**Comment by BaraMGB** [comment]
Okay, one question before it is to late. There were a lot of requests in the past for change the size of the instrument window. Shouldn't we think about this issue before the instruments are redesigned? 

Wanna link this issue https://github.com/LMMS/lmms/issues/3067

**Comment by RebeccaLaVie** [comment]
**A couple things that I think should be discussed regarding the instruments:**

- Changing the size of the plugins would help to bring the design closer to the single window concept. What size would we want to change it to? Should the height have a fixed size, but the width depend on the instrument?
- It reduces and simplifies the code if we leverage the stock knobs. I know there were a lot of requests to be able to choose a custom knob color, but this might be worth the work to simplify things.
- Tres mentioned that he thinks all instruments that use them should carry common size and shape 2D controller pad/and or spectral output, and personally, I would like to see this happen where possible.
- What other aspects of these designs can be standardized?

![screenshot_20170506-150400](https://cloud.githubusercontent.com/assets/7960169/25821591/9b34ac40-33ea-11e7-89ae-8610ac915c8e.png)

**Comment by Umcaruje** [comment]
>Pixmap are difficult to handle this way.

Not really, you'd need to just have double sized renders for hidpi. There's just some stuff you can't do without pixmaps.

**Comment by Mark-Agent003** [comment]
@RebeccaLaVie,

> By this, I mean that anyone can use any of the colors in the palette above and a custom logo to slightly distinguish each plugin, because this has been requested by the community. We still want the knobs, colors, and style to stay the same.

Yes, I understand. 😄
Also, I agree we need to change the instrument dimensions at some point, and, in my opinion, the height should be fixed, and the width should be variable.

**Comment by RebeccaLaVie** [comment]
@Texxiliator @IvanMaldonado @Umcaruje @BaraMGB @Mark-Agent003 @JordanKyser 

Okay, so after a discussion with many of you on Discord, this is the direction we are going to take with the redesigns. We are going to use the same layout for the instruments as the single window concept, with the same colors and design elements of our current theme. We will be using the same design for all of the knobs, with the exception that you can change the color to match the logo. The logo will be placed in the upper left corner of all of the plugins. All controls will be natively themed.

This project will require collaboration between the designers and coders that make themselves available. Right now, we are going to focus on finishing 1.2 before we work on the instruments.

**Comment by BaraMGB** [comment]
These are the layouts we want to use. Sorry for the quality. I compressed it. 

 - We want to use the layout that means fixed in height but width can be different. 
 - the designer can use any Color from the modern noir palette (see above)

![budislav_instruments](https://cloud.githubusercontent.com/assets/6502580/25826833/ab09cfde-3448-11e7-8cc1-da0daea40026.jpg)

**Comment by Mark-Agent003** [comment]
@RebeccaLaVie, in my understanding, we are only supposed to change color on the knobs to match the logo, and every other color should be either the green/gray/black in the theme?

**Comment by Mark-Agent003** [comment]
@RebeccaLaVie 
![image](https://cloud.githubusercontent.com/assets/25230510/25974931/1d3d7356-3671-11e7-83f1-0d869d5745de.png)
(rough draft)
Is this the style you were looking for? I used the dimensions from the single window concept.

**Comment by Texxiliator** [comment]
@Mark-Agent003 she is not going to reply in some time. She is on a break.

**Comment by RebeccaLaVie** [comment]
@Texxiliator @Mark-Agent003 

> she is not going to reply in some time

I'll try to reply to everything on my next weekend. 

> She is on a break

I'm just really busy right now with work.

:blush:

**Comment by Texxiliator** [comment]
@RebeccaLaVie oh
Deseiver told me you are.


(do you know that there is worst of lmms?)

**Comment by Texxiliator** [comment]
@RebeccaLaVie so is @Mark-Agent003 style the thing you are looking for?

**Comment by BaraMGB** [comment]
@Texxiliator can you please stop rushing the people? This isn't a chat channel here. Slow down a little bit!

**Comment by Texxiliator** [comment]
@BaraMGB alright, sorry.

**Comment by JordanKyser** [comment]
If someone else wants to redesign ZynAddSubFX, they can do it. I don't have much spare time to do it (even if it is for a couple months). ☹️ 
I am very sorry for the inconvenience.

**Comment by RebeccaLaVie** [comment]
@Texxiliator I appreciate your enthusiasm, but I'm going to need just a bit more patience. :+1: 

@Mark-Agent003 Perhaps the contrast could be increased and the design could benefit from being lighter, but yes, I believe that mockup is headed in the right direction.

I think an important question to ask is if we want the height to be the same as it is in the single window design? That size might not be ideal for our current layout.

@JordanKyser That's no problem.

**Comment by BaraMGB** [comment]
In my understanding we do not redesign the instruments for 1.2. That would mean that we had to rewrite the layout of the complete instrument window. I guess it were incompatible to the current instrument window. With 1.2 we should make a point and think for the future. I am very confident that we can have a single window GUI for 1.3 but redesign the instruments for 1.2 layout is a waste of time if we want the single window GUI.

**Comment by RebeccaLaVie** [comment]
@BaraMGB I'm sorry if I confused everything be responding to the feedback here.

When I said 

> That size might not be ideal for our current layout.

I just didn't want people to start mocking up ideas before the size has been finalized. Whether the single window concept can be fully realized at 1.3 or not, we still need to think about what height would be best for the ui when we get there, and especially before any mockups are made.


> In my understanding we do not redesign the instruments for 1.2. 


Yes, I agree that this is not a project for 1.2. Let's come back to this thread when we're ready to work on it :+1:

**Comment by BaraMGB** [comment]
> Let's come back to this thread when we're ready to work on it 

Yes, I'll make some mockups and try to start envelope a concept for the single window GUI. I'll open a new issue for this when I have some results.

**Comment by budislav** [comment]
Guys I already done the math. That height is perfect when you consider all dimensions of UI elements and empty spaces inside of all effects.
If you use a ruler there you can notice a pattern in those dimensions. If you take a smaller height you can't put everything inside nicely, if you make bigger that would be a problem for smaller screens.
It takes me a lot of time to finish everything, I don't see a reason why someone would spend half year to make everything different now, instead of start coding **exactly this design**.
You have around 10 effects and 24 complex instruments or other parts, so good luck and keep your eye on balance between UI elements.

**Comment by RebeccaLaVie** [comment]
@budislav the point of my efforts here is to literally translate your work on the effects plugins into the UI as much as possible with our current capabilities. Despite a few miscomunications on these redesigns, I do not intend to reinvent the wheel.

**Comment by budislav** [comment]
@RebeccaLaVie 
> the point of my efforts in this thread are to literally translate your work on the effects plugins into the UI as much as possible with our current capabilities. 

I am grateful to you :)

I have a task for you if you want. You said:

> The logo will be placed in the upper left corner of all of the plugins

Make an icon for every instrument in LMMS. Zynaddsubfx only have new icon I made, two triangles you can see in new Zyn Fusion concept so you should insert that one. Icons should be 16x16px like in song editor mono. Make it pixel perfect, vector off course. That is really hard task for 16x16px, I avoided it but someone need to make it :)
There is no need for effect icons because there is no space there really and there are hundreds of LADSPA and others..
Take as long as you need, don't rush, I have two more hard tasks for you I avoided, and I am not the right guy for them :D

**Comment by RebeccaLaVie** [comment]
@budislav I guess that makes it my turn to fire back a task for you :D 

Most of the people that will be working on the plugin designs use Inkscape. I'm assuming that your sources files are .ai. It would be much easier to integrate your work on these if you provided us with .svg versions of the mockups that we referenced above.

**Comment by budislav** [comment]
@RebeccaLaVie No problem, I will convert everything to .svg including all parts of UI. I need some time because letters, icons and knobs needs to redraw again.
